containers/src/Kokkos_ScatterView.hpp

```diff
@@ -144,39 +151,277 @@ struct DefaultContribution<Kokkos::Cuda, Kokkos::Experimental::ScatterDuplicated
 };
 #endif
 
-/* ScatterValue is the object returned by the access operator() of ScatterAccess,
-   similar to that returned by an Atomic View, it wraps Kokkos::atomic_add with convenient
-   operator+=, etc. */
+/* ScatterValue <Op=ScatterSum, contribution=ScatterNonAtomic> is the object returned by the access operator() of ScatterAccess,
+   This class inherits from the Sum<> reducer and it wraps join(dest, src) with convenient operator+=, etc. 
+   Note the addition of update(ValueType const& rhs) and reset()  so that all reducers can have common functions
```

Comment:
Why not have all reducers implement `operator()`, instead of giving the reduction method different names in different classes?

Reply:
I think the reasoning is to make the code look the same as a serial scatter implementation into a "normal" View or even an array. It should make it easier for people to port codes based on plain C arrays and `+=`, and makes the reduction operator clear at that call site instead of hiding it in the type.